Content.Server/Objectives/Systems/KillPersonConditionSystem.cs

```diff
@@ -42,29 +41,15 @@ private void OnGetProgress(EntityUid uid, KillPersonConditionComponent comp, ref
 
     private void OnPersonAssigned(EntityUid uid, PickRandomPersonComponent comp, ref ObjectiveAssignedEvent args)
     {
-        // invalid objective prototype
-        if (!TryComp<TargetObjectiveComponent>(uid, out var target))
-        {
-            args.Cancelled = true;
-            return;
-        }
-
-        // target already assigned
-        if (target.Target != null)
-            return;
-
-        // no other humans to kill
-        var allHumans = _mind.GetAliveHumansExcept(args.MindId);
-        if (allHumans.Count == 0)
-        {
-            args.Cancelled = true;
-            return;
-        }
-
-        _target.SetTarget(uid, _random.Pick(allHumans), target);
+        AssignRandomTarget(uid, args, false);
     }
 
     private void OnHeadAssigned(EntityUid uid, PickRandomHeadComponent comp, ref ObjectiveAssignedEvent args)
+    {
+        AssignRandomTarget(uid, args, true);
+    }
+
+    private void AssignRandomTarget(EntityUid uid, ObjectiveAssignedEvent args, bool headsOnly)
```

Comment:
Rather than a bool to filter for heads, I wonder if the caller could pass in a Predicate to be used as a filter.

Reply:
It's doable and a Predicate would probably end up being more flexible, but also feels way less readable, so not sure if it'd be preferred